### PR TITLE
fix(http): Set the transfer-encoding to chunked mode only if the clie…

### PR DESCRIPTION
**Pull request by brasseld**
…nt is pushing data to the gateway

Closes gravitee-io/issues#1655

### CODE REVIEW

File: gravitee-gateway-standalone/gravitee-gateway-standalone-container/src/test/java/io/gravitee/gateway/standalone/PostContentGatewayTest.java [hunk 1]
```diff
@@ -148,4 +148,29 @@ public void call_case3_raw() throws Exception {
             assertTrue(false);
         }
     }
+
```
Comment (aesteve) [inline gravitee-gateway-standalone/gravitee-gateway-standalone-container/src/test/java/io/gravitee/gateway/standalone/PostContentGatewayTest.java]: Maybe test that a request with empty body but no Transfer-Encoding header is forwarded to upstream without being chunked (basically : please add also the opposite of your current test) ?

File: gravitee-gateway-http/src/main/java/io/gravitee/gateway/http/connector/VertxProxyConnection.java [hunk 2]
```diff
@@ -94,6 +95,15 @@ public void handleConnectTimeout(Throwable throwable) {
     @Override
     public VertxProxyConnection write(Buffer chunk) {
```
Comment (aesteve) [inline gravitee-gateway-http/src/main/java/io/gravitee/gateway/http/connector/VertxProxyConnection.java]: I can't ensure that deffering the write of headers wouldn't cause any regression though. (hope tests would catch any). but lgtm

File: gravitee-gateway-standalone/gravitee-gateway-standalone-container/src/test/java/io/gravitee/gateway/standalone/PostContentGatewayTest.java (hunk 1 above)
Comment (brasseld) [reply gravitee-gateway-standalone/gravitee-gateway-standalone-container/src/test/java/io/gravitee/gateway/standalone/PostContentGatewayTest.java]: Added

Comment (aesteve) [reply gravitee-gateway-standalone/gravitee-gateway-standalone-container/src/test/java/io/gravitee/gateway/standalone/PostContentGatewayTest.java]: lgtm

**Pull request merged**